site.yml

```diff
@@ -23,6 +23,7 @@
 - hosts: all
   roles:
     - docker
+    - { role: weave, when: weave_enabled|bool }
```

Comment:
agree on the big PRs - think it will probably affect the semantics of this PR though so probably just worth agreeing now and getting this in as is...

Rather than having weave_enabled / calico_enabled / flannel_enabled, would it be a bit nicer if we had a 'network_plugin' role which we could just set ```network_plugin: flannel``` rather than a load of disconnected roles and varying toggle switches